woocommerce/woocommerce
# add/lys xstate scaffolding

- dev: added xstate 5
- add xstate scaffolding for launch your store

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/45116.

Note that this PR has installed xstate v5 simultaneously alongside xstate v4. There is a postinstall script which correctly symlinks the expected packages. 

This arrangement will be required to be in place until we migrate the previous xstate machines over to v5.




### How to test the changes in this Pull Request:

Please exclude from Solaris and GlobalStep testing as this is gated behind a feature flag and is part of the Launch Your Store feature which will be tested as a complete feature when the call for testing is released.

1. Ensure that the "launch-your-store" feature flag is enabled using WC Beta Tester plugin
2. Go to http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Flaunch-your-store and test around

https://github.com/woocommerce/woocommerce/assets/27843274/59d5b74d-f43f-4d75-91d2-e4c33ed52681

Note that the browser back button will not work as expected as it has not been implemented and will be implemented in a follow up PR.

The XState inspector will also not work as the currently installed version only works for xstate v4 while this PR uses xstate v5. This will also be addressed in a follow up PR. In the mean time please use the debug output in this PR to assist.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
